src/utils/array.js

```diff
@@ -101,12 +101,33 @@ export function fromIterator(iterator) {
  *
  * @param {Array<T>} array
  * @param {T} item
+ * @return {boolean}
  * @template T
  */
 export function pushIfNotExist(array, item) {
   if (array.indexOf(item) < 0) {
     array.push(item);
+    return true;
   }
+  return false;
+}
+
+/**
+ * Removes one item in the array. The array is assumed to have unique values
+ * only. Returns `true` if the array changed.
+ *
+ * @param {!Array<T>} array
+ * @param {T} item
+ * @return {boolean}
+ * @template T
+ */
+export function removeUniqueItem(array, item) {
```

Comment:
This can be named `removeItem` with the description that it removes the first instance of the given item in the array. That is what it is doing anyway, and there is nothing asserting that the item is unique within the array.